pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1436,51 +1446,92 @@ func (c *Cloud) InstanceType(ctx context.Context, nodeName types.NodeName) (stri
 }
 
 // getCandidateZonesForDynamicVolume retrieves  a list of all the zones in which nodes are running
-// It currently involves querying all instances
 func (c *Cloud) getCandidateZonesForDynamicVolume() (sets.String, error) {
 	// We don't currently cache this; it is currently used only in volume
 	// creation which is expected to be a comparatively rare occurrence.
 
 	// TODO: Caching / expose v1.Nodes to the cloud provider?
 	// TODO: We could also query for subnets, I think
+	zones, err := c.getCachedCandidateZonesForDynamicVolume()
+	if err != nil {
+		glog.V(2).Infof("getCachedCandidateZonesForDynamicVolume() failed, falling back to legacy method: %v", err)
 
-	filters := []*ec2.Filter{newEc2Filter("instance-state-name", "running")}
+		filters := []*ec2.Filter{newEc2Filter("instance-state-name", "running")}
 
-	instances, err := c.describeInstances(filters)
+		instances, err := c.describeInstances(filters)
```

Comment:
This call is really not optimal (it gets all instances in the account and filter locally)
I wonder if we shouldn't modify this function (describeInstances) instead (as the TODO shows in the existing code)
It's a bit of a detail that is probably better discussed upstream

Reply:
This is the legacy path, we don't use it if the Node-Informer solution works. Maybe this path could be completely removed if the assumption is the Node-Informer should always work. That's the discussion I hope to see upstream, indeed.